src/custom/state/enhancedTransactions/reducer.ts

```diff
@@ -0,0 +1,34 @@
+import { AnyAction, createReducer } from '@reduxjs/toolkit'
+import transactionsReducer, { initialState, TransactionState } from 'state/transactions/reducer'
+import { cancelTransaction, replaceTransaction } from 'state/enhancedTransactions/actions'
+
+export const reducer = createReducer(initialState, (builder) =>
+  builder
+    .addCase(cancelTransaction, (transactions, { payload: { chainId, hash } }) => {
+      if (!transactions[chainId]?.[hash]) {
+        console.error('Attempted to cancel an unknown transaction.')
+        return
+      }
+      const allTxs = transactions[chainId] ?? {}
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { [hash]: omit, ...txs } = allTxs
```

Comment:
Would it be all this code equivalent to just do a delete on the hash? 
My thinking is that `@reduxjs/toolkit` lets u mutate the object
`delete allTxs[hash]`